src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -24,50 +28,86 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.util")
-)
+	log = golog.LoggerFor("flashlight.proxied")
 
-// HTTPFetcher is a simple interface for types that are able to fetch data over HTTP.
-type HTTPFetcher interface {
-	Do(*http.Request) (*http.Response, error)
-}
+	proxyAddrMutex sync.RWMutex
+	proxyAddr      = eventual.DefaultUnsetGetter()
+
+	// ErrChainedProxyUnavailable indicates that we weren't able to find a chained
+	// proxy.
+	ErrChainedProxyUnavailable = errors.New("chained proxy unavailable")
+)
 
 func success(resp *http.Response) bool {
 	return resp.StatusCode > 199 && resp.StatusCode < 400
 }
 
-// NewChainedAndFronted creates a new struct for accessing resources using chained
-// and direct fronted servers in parallel.
-func NewChainedAndFronted(proxyAddrFN eventual.Getter) *chainedAndFronted {
-	cf := &chainedAndFronted{proxyAddrFN: proxyAddrFN}
+// SetProxyAddr sets the eventual.Getter that's used to determine the proxy's
+// address. This MUST be called before attempting to use the proxied package.
+func SetProxyAddr(addr eventual.Getter) {
```

Comment:
Correct, it never changes after being set the first time.

-- 
Cheers,
Ox

PGP Key - https://pgp.mit.edu/pks/lookup?op=get&search=ox@getlantern.org

-----------------------------------------------------------------------------------------

"I love people who harness themselves, an ox to a heavy cart,
who pull like water buffalo, with massive patience,
who strain in the mud and the muck to move things forward,
who do what has to be done, again and again."

- Marge Piercy